packages/wasm-parser/test/custom-sections.js

```diff
@@ -0,0 +1,556 @@
+              startOffset: 34,
+              size: {
+                type: "NumberLiteral",
+                value: 2,
+                raw: "2",
+                loc: {
+                  start: {
+                    line: -1,
+                    column: 34
+                  },
+                  end: {
+                    line: -1,
+                    column: 35
+                  }
+                }
+              },
+              vectorOfSize: {
+                type: "NumberLiteral",
+                value: 1,
+                raw: "1",
+                loc: {
+                  start: {
+                    line: -1,
+                    column: 35
+                  },
+                  end: {
+                    line: -1,
+                    column: 36
+                  }
+                }
+              }
+            },
+            {
+              type: "SectionMetadata",
+              section: "custom",
+              startOffset: 38,
+              size: {
+                type: "NumberLiteral",
+                value: 155,
+                raw: "155",
+                loc: {
+                  start: {
+                    line: -1,
+                    column: 38
+                  },
+                  end: {
+                    line: -1,
+                    column: 40
+                  }
+                }
+              }
+            },
+            {
+              type: "SectionMetadata",
+              section: "code",
+              startOffset: 196,
+              size: {
+                type: "NumberLiteral",
+                value: 4,
+                raw: "4",
+                loc: {
+                  start: {
+                    line: -1,
+                    column: 196
+                  },
+                  end: {
+                    line: -1,
+                    column: 197
+                  }
+                }
+              },
+              vectorOfSize: {
+                type: "NumberLiteral",
+                value: 1,
+                raw: "1",
+                loc: {
+                  start: {
+                    line: -1,
+                    column: 197
+                  },
+                  end: {
+                    line: -1,
+                    column: 198
+                  }
+                }
+              }
+            },
+            {
+              type: "SectionMetadata",
+              section: "custom",
+              startOffset: 202,
+              size: {
+                type: "NumberLiteral",
+                value: 10,
+                raw: "10",
+                loc: {
+                  start: {
+                    line: -1,
+                    column: 202
+                  },
+                  end: {
+                    line: -1,
+                    column: 203
+                  }
+                }
+              }
+            }
+          ]
+        }
+      }
+    ]
+  };
+}
+
+function expectedForCustomAfterName() {
```

Comment:
Could we use a JSON string for the AST instead? You can then compare it (line by line with good errors) using https://github.com/xtuc/webassemblyjs/blob/master/packages/helper-test-framework/src/index.js#L35

Reply:
Ok I see, in that case I would recommend you using `traverse` and target the nodes directly.